### PR TITLE
[sovereign] Sync acc+validator dbs

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- For sovereign `SyncBlock` , added syncers for both user accounts and validator accounts
- Injected validator accounts db syncer factory in runType comps
  
## Proposed changes
- Moved `ValdiatorDBSyncer` and `syncAccountsDBs` from `metaBlock` sync to `baseSync` to be used for sovereign as well

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: process/sync/baseSync.go [hunk 1]
```diff
@@ -1238,6 +1243,29 @@ func (boot *baseBootstrap) handleTokensSuppliesRepopulation() error {
 	return tokensSuppliesProc.SaveSupplies()
 }
 
+func (boot *baseBootstrap) syncAccountsDBs(key []byte, id string) error {
+	// TODO: refactor this in order to avoid treatment based on identifier
```
Comment (sasurobert) [inline process/sync/baseSync.go:1247]: I do not think this is a huge problem - as it only starts the sync - thus this check is done only once.

File: process/sync/baseSync.go [hunk 2]
```diff
@@ -1238,6 +1243,29 @@ func (boot *baseBootstrap) handleTokensSuppliesRepopulation() error {
...
+	switch id {
+	case dataRetriever.UserAccountsUnit.String():
+		log.Error("UserAccountsUnit")
```
Comment (sasurobert) [inline process/sync/baseSync.go:1250]: no log.Error please.

File: process/sync/baseSync.go [hunk 3]
```diff
@@ -1238,6 +1243,29 @@ func (boot *baseBootstrap) handleTokensSuppliesRepopulation() error {
...
+		return boot.syncUserAccountsState(key)
+	case dataRetriever.PeerAccountsUnit.String():
+		log.Error("PeerAccountsUnit")
```
Comment (sasurobert) [inline process/sync/baseSync.go:1253]: no log.Error please.

File: process/sync/baseSync.go [hunk 4]
```diff
@@ -1238,6 +1243,29 @@ func (boot *baseBootstrap) handleTokensSuppliesRepopulation() error {
...
+		return fmt.Errorf("invalid trie identifier, id: %s", id)
+	}
+}
+
+func (boot *baseBootstrap) syncShardAccountsDBs(key []byte, _ string) error {
+	return boot.syncUserAccountsState(key)
+}
+
+func (boot *baseBootstrap) syncValidatorAccountsState(key []byte) error {
+	log.Warn("base sync: started syncValidatorAccountsState")
```
Comment (sasurobert) [inline process/sync/baseSync.go:1265]: no log warn.

File: process/mock/runTypeComponentsStub.go [hunk 5]
```diff
@@ -9,36 +9,39 @@ import (
...
+		AdditionalStorageServiceFactory:            &testscommon.AdditionalStorageServiceFactoryMock{},
+		ShardCoordinatorFactory:                    sharding.NewMultiShardCoordinatorFactory(),
+		NodesCoordinatorWithRaterFactory:           nodesCoordinator.NewIndexHashedNodesCoordinatorWithRaterFactory(),
+		RequestHandlerFactory:                      requestHandlers.NewResolverRequestHandlerFactory(),
+		AccountCreator:                             &stateMock.AccountsFactoryStub{},
+		OutGoingOperationsPool:                     &sovereignMocks.OutGoingOperationsPoolMock{},
+		DataCodec:                                  &sovereignMocks.DataCodecMock{},
+		TopicsChecker:                              &sovereignMocks.TopicsCheckerMock{},
+		RequestersContainerFactoryCreatorField:     requesterscontainer.NewShardRequestersContainerFactoryCreator(),
+		ValidatorAccountsSyncerFactoryHandlerField: syncerFactory.NewValidatorAccountsSyncerFactory(),
```
Comment (axenteoctavian) [inline process/mock/runTypeComponentsStub.go:44]: Maybe use mock here ValidatorAccountsSyncerFactoryMock?

File: process/mock/runTypeComponentsStub.go [hunk 6]
```diff
@@ -48,15 +51,16 @@ func NewSovereignRunTypeComponentsStub() *RunTypeComponentsStub {
...
+		AdditionalStorageServiceFactory:            &testscommon.AdditionalStorageServiceFactoryMock{},
+		ShardCoordinatorFactory:                    sharding.NewSovereignShardCoordinatorFactory(),
+		NodesCoordinatorWithRaterFactory:           &testscommon.NodesCoordinatorFactoryMock{},
+		RequestHandlerFactory:                      requestHandlerFactory,
+		AccountCreator:                             &stateMock.AccountsFactoryStub{},
+		OutGoingOperationsPool:                     &sovereignMocks.OutGoingOperationsPoolMock{},
+		DataCodec:                                  &sovereignMocks.DataCodecMock{},
+		TopicsChecker:                              &sovereignMocks.TopicsCheckerMock{},
+		RequestersContainerFactoryCreatorField:     requesterscontainer.NewSovereignShardRequestersContainerFactoryCreator(),
+		ValidatorAccountsSyncerFactoryHandlerField: syncerFactory.NewSovereignValidatorAccountsSyncerFactory(),
```
Comment (axenteoctavian) [inline process/mock/runTypeComponentsStub.go:63]: Maybe use mock here ValidatorAccountsSyncerFactoryMock?

File: process/sync/baseSync.go (hunk 1 above)
Comment (mariusmihaic) [reply process/sync/baseSync.go:1247]: Right, this was copy-pasted from original meta code, wanted to leave the TODO as it was

File: process/sync/baseSync.go (hunk 2 above)
Comment (mariusmihaic) [reply process/sync/baseSync.go:1250]: Right, deleted it

File: process/sync/baseSync.go (hunk 3 above)
Comment (mariusmihaic) [reply process/sync/baseSync.go:1253]: Right, deleted it

File: process/sync/baseSync.go (hunk 4 above)
Comment (mariusmihaic) [reply process/sync/baseSync.go:1265]: This was actually copy-pasted from original code when syncing data on meta, wanted to leave it as it is.

File: factory/consensus/consensusComponents_test.go [hunk 7]
```diff
@@ -189,8 +190,9 @@ func createMockConsensusComponentsFactoryArgs() consensusComp.ConsensusComponent
 					return &processMock.BootstrapperStub{}, nil
 				},
 			},
-			ShardMessengerFactoryField: &factoryMocks.ShardChainMessengerFactoryMock{},
-			ConsensusModelType:         consensus.ConsensusModelV1,
+			ShardMessengerFactoryField:                 &factoryMocks.ShardChainMessengerFactoryMock{},
+			ValidatorAccountsSyncerFactoryHandlerField: factory.NewValidatorAccountsSyncerFactory(),
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply factory/consensus/consensusComponents_test.go:194]: Can't, tests would fail

File: process/mock/runTypeComponentsStub.go (hunk 5 above)
Comment (mariusmihaic) [reply process/mock/runTypeComponentsStub.go:44]: This "stub" is not a stub actually, it uses mostly real comps;
Also, got the same "cyclic" import issue here, there is a todo to fix it

File: process/mock/runTypeComponentsStub.go (hunk 6 above)
Comment (mariusmihaic) [reply process/mock/runTypeComponentsStub.go:63]: This "stub" is not a stub actually, it uses mostly real comps;
Also, got the same "cyclic" import issue here, there is a todo to fix it

**Review by sasurobert — APPROVED** [review]

**Pull request merged**